File: src/main/java/demo/bfims/Entities/Inventory/LiteraryPiece.java
package demo.bfims.Entities.Inventory;

import demo.bfims.Enums.LiteraryType;
import demo.bfims.Enums.ItemType;
import demo.bfims.Enums.PublicationItemType;
import jakarta.persistence.*;

@Entity
public class LiteraryPiece extends PublicationItem {
    @Enumerated(EnumType.STRING)
    LiteraryType type;

    public LiteraryPiece() {
        this.setPublicationItemType(PublicationItemType.LITERARY_PIECE);
    }

    public LiteraryType getType() {
        return type;
    }
    public void setType(LiteraryType type) {
        this.type = type;
    }

    @Override
    public String toString() {
        return "LiteraryPiece{" +
                "type=" + type +
                ", publication=" + publication +
                "} " + super.toString();
    }
}

File: frontend/src/app/DTOs/Inventory/LiteraryPiece.ts
import {PublicationItem} from './PublicationItem';
import {PublicationItemType} from '../../Enums/PublicationItemType';
import {LiteraryType} from '../../Enums/LiteraryType';

export class LiteraryPiece extends PublicationItem {
  public literaryType: LiteraryType;

  constructor(...data: any) {
    data = data[0];
    super(data);
    if (data) {
      this.literaryType = data["literaryType"] || null;
    }
    this.publicationItemType = PublicationItemType.LITERARY_PIECE;
  }
}
